internal/locate/region_cache.go

```diff
@@ -1203,6 +1204,149 @@ func (c *RegionCache) LocateKeyRange(bo *retry.Backoffer, startKey, endKey []byt
 	return res, nil
 }
 
+type BatchLocateKeyRangesOption struct {
+	needLeader  bool // whether load leader only, if it's set to true, regions without leader will be skipped.
```

Comment:
`needRegionWithValidLeader` is more understandable.

Reply:
The leader is not empty even it's an invalid or outdated leader. For a new splitted regions, they are reported to PD after leaders are elected, so the leaders are also not nil.

So this parameter won't affect anything from what I know, just keep it to keep the compatibility.

I think `needRegionWithLeaderPeer` a proper name.